funccount.py

```diff
@@ -0,0 +1,84 @@
+import os
+import re
+import shutil
+import signal
+import subprocess as sp
+import threading
+import time
+
+class FunccountInvalidFunctionError(Exception):
+    pass
+
+class FunccountNotStartedError(Exception):
+    pass
+
+class Funccount:
+    def _sleep_and_run(self):
+        self.event.wait(self.sleep)
+
+        self.lock.acquire()
```

Comment:
The preferred syntax is `with self.lock`.... But this locking stuff seems awfully complicated.

And:

 * Probably shouldn't start the process if the "event" is terminated early
 * No need to lock or try to race, just stop the event and join the thread; creating the process is fast.
 * Should pass any exceptions from the thread to the main thread
 * I usually find it is better to subclass Thread than use a specific method like this.
 
So roughly something like:

```python
class FuncCount(threading.Thread):
    def __init__(self, funcs, sleep=0, executable="./funccount2"):
        super().__init__()
        self._cmd = [executable, funcs]
        self._sleep = sleep

        self._stop_event = threading.Event()
        self._proc = None
        self._excp = None

    def run(self):
        if self._stop_event.wait(self._sleep):
            return

        try:
            self._proc = subprocess.Popen(self._cmd, text=True,
                                          stdout=subprocess.PIPE,
                                          stderr=subprocess.PIPE,
                                          start_new_session=True)
        except Exception as e:
            self._excp = e

    def stop(self):
        self._stop_event.set()
        self.join()

        if self._excp:
            raise self._excp

        if not self._proc:
            return

        pgid = os.getpgid(self._proc.pid)
        os.killpg(pgid, signal.SIGINT)

        out, err = self._proc.communicate(timeout=1)
        #etc

```